tests/consensus/test_soft_voided3.py

```diff
@@ -44,8 +44,12 @@ def _run_test(self, simulator, soft_voided_tx_ids):
         gen_tx2 = simulator.create_tx_generator(manager2, rate=10 / 60., hashpower=1e6, ignore_no_funds=True)
         gen_tx2.start()
 
-        while not gen_tx2.latest_transactions:
+        for _ in range(100):
+            if gen_tx2.latest_transactions:
+                break
             simulator.run(300)
+        else:
+            self.fail('took too long to generate a transaction')
```

Comment:
Replace this block by `gen_tx2.run_until_find_one_transaction()`.

Reply:
Done.

Reply:
Well done. Thanks!